pkg/segment/aggregations/evalaggs.go

```diff
@@ -23,297 +23,543 @@ import (
+			err := PopulateFieldToValueFromSegStats(fields, measureAgg, sstMap, fieldToValue, i)
 			if err != nil {
-				return fmt.Errorf("ComputeAggEvalForRange: can not get the float value: %v", err)
+				return fmt.Errorf("ComputeAggEvalForRange: Error while populating fieldToValue from sstMap, err: %v", err)
 			}
-			// Keep maximum and minimum values
-			if eValFloat < minVal {
-				minVal = eValFloat
-			}
-			if eValFloat > maxVal {
-				maxVal = eValFloat
+
+			rangeStat, err = PerformEvalAggForRange(measureAgg, exists, rangeStat, fieldToValue)
+			exists = true
+			if err != nil {
+				return fmt.Errorf("ComputeAggEvalForRange: Error while performing eval agg for range, err: %v", err)
 			}
 		}
 	}
 
-	rangeStat := &structs.RangeStat{}
-	rangeStatVal, exists := runningEvalStats[measureAgg.String()]
-	if !exists {
-		rangeStat.Min = minVal
-		rangeStat.Max = maxVal
-	} else {
-		rangeStat = rangeStatVal.(*structs.RangeStat)
-		if rangeStat.Min > minVal {
-			rangeStat.Min = minVal
-		}
-		if rangeStat.Max < maxVal {
-			rangeStat.Max = maxVal
-		}
-	}
-	runningEvalStats[measureAgg.String()] = rangeStat
+	runningEvalStats[measureAgg.String()] = &rangeStat
 	rangeVal := rangeStat.Max - rangeStat.Min
 
-	enclosure, exists := measureResults[measureAgg.String()]
-	if !exists {
-		enclosure = utils.CValueEnclosure{
-			Dtype: utils.SS_DT_FLOAT,
-			CVal:  rangeVal,
-		}
-		measureResults[measureAgg.String()] = enclosure
+	enclosure := utils.CValueEnclosure{
+		Dtype: utils.SS_DT_FLOAT,
+		CVal:  rangeVal,
 	}
+	measureResults[measureAgg.String()] = enclosure
 
-	eValFloat, err := enclosure.GetFloatValue()
+	return nil
+}
+
+func GetFloatValueAfterEvaluation(measureAgg *structs.MeasureAggregator, fieldToValue map[string]utils.CValueEnclosure) (float64, string, bool, error) {
+	valueStr, err := measureAgg.ValueColRequest.EvaluateToString(fieldToValue)
+	if err != nil {
+		return 0, "", false, fmt.Errorf("GetFloatValueAfterEvaluation: Error while evaluating eval function: %v", err)
+	}
+	floatVal, err := dtypeutils.ConvertToFloat(valueStr, 64)
 	if err != nil {
-		return fmt.Errorf("ComputeAggEvalForRange: Attempted to perform aggregate min(), but the column %s is not a float value", fields[0])
+		return 0, valueStr, false, nil
 	}
+	return floatVal, valueStr, true, nil
+}
 
-	if eValFloat < rangeVal {
-		enclosure.CVal = rangeVal
-		measureResults[measureAgg.String()] = enclosure
+func PopulateFieldToValueFromSegStats(fields []string, measureAgg *structs.MeasureAggregator, sstMap map[string]*structs.SegStats, fieldToValue map[string]utils.CValueEnclosure, i int) error {
+	for _, field := range fields {
+		sst, ok := sstMap[field]
+		if !ok {
+			return fmt.Errorf("PopulateFieldToValueFromSegStats: applyAggOpOnSegments sstMap was nil for aggCol %v", measureAgg.MeasureCol)
+		}
+
+		if i >= len(sst.Records) {
+			return fmt.Errorf("PopulateFieldToValueFromSegStats: Incorrect length of field: %v for aggCol: %v", field, measureAgg.String())
```

Comment:
For the `aggCol`, why do you specify it as `measureAgg.MeasureCol` in one log but as `measureAgg.String()` in another?

Reply:
I used the existing comment. They technically refer to the same thing i.e. the field's name used in the measure function. I will update it to make it consistent.

Reply:
addressed in the latest commit.